Hunting Queries/SigninLogs/InactiveAccounts.yaml

```diff
@@ -19,30 +19,57 @@ query: |
+  // also always keep InvestigationPriority if 1 or more
+  | where (not(ActivityInsights.FirstTimeUserUsedApp == 'True' and ActivityInsights.AppUncommonlyUsedAmongPeers == 'False') or InvestigationPriority > 0)
+  | where (not(ActivityInsights.FirstTimeUserConnectedViaBrowser == 'True' and ActivityInsights.BrowserUncommonlyUsedAmongPeers == 'False') or InvestigationPriority > 0)
+  | where (not(ActivityInsights.FirstTimeUserAccessedResource == 'True' and ActivityInsights.ResourceUncommonlyUsedAmongPeers == 'False') or InvestigationPriority > 0)
+  // for ISP, it makes more sense to exclude if Uncommon in Tenant or Uncommon among peers is false.
+  | where (not(ActivityInsights.FirstTimeUserConnectedViaISP == 'True' and (ActivityInsights.ISPUncommonlyUsedInTenant == 'False' or ActivityInsights.ISPUncommonlyUsedAmongPeers == 'False')) or InvestigationPriority > 0)
+  | extend UEBA_Insights = pack_dictionary("TimeGenerated", TimeGenerated, "ActivityInsights", ActivityInsights, "UsersInsights", UsersInsights, "DevicesInsights", DevicesInsights)
+  | summarize UEBA_ItemIds = make_set(_ItemId), UEBA_SourceRecordIds = make_set(SourceRecordId), UEBA_Insights = make_set(UEBA_Insights) by
+  UEBA_UserPrincipalName = UserPrincipalName, JoinedWithType = Type, UEBA_ActionType = ActionType, UEBA_SourceIPAddress = SourceIPAddress, UEBA_SourceIPLocation = SourceIPLocation, UEBA_InvestigationPriority = InvestigationPriority
+  | extend UEBA_Info = pack_dictionary("UEBA_Insights", UEBA_Insights, "UEBA_ItemIds", UEBA_ItemIds, "UEBA_SourceRecordIds", UEBA_SourceRecordIds)
+  | project-away UEBA_ItemIds, UEBA_SourceRecordIds, UEBA_Insights
+  | join kind=inner ( 
+    SigninsSummary
+  ) on $left.UEBA_UserPrincipalName == $right.UserPrincipalName, $left.UEBA_SourceIPAddress == $right.IPAddress
+  | project-reorder StartTime, EndTime, UserPrincipalName, UserId, IPAddress, UserType, loginCountToday, JoinedWithType
+  ;
+  SigninsWithUEBA 
   | join kind= leftanti (
-     // filter out newly created user accounts
+     // filter out newly created user accounts from last 7 days
      AuditLogs
-     | where TimeGenerated between(midtime..starttime)
+     | where TimeGenerated between(midtime..endtime)
      | where OperationName == "Add user"
-     // Normalize to lower case in order to match against equivalent UPN in Signin logs
-     | extend NewUserPrincipalName = tolower(extractjson("$.userPrincipalName", tostring(TargetResources[0]), typeof(string)))
-  ) on $left.UserPrincipalName == $right.NewUserPrincipalName
+     | summarize by NewUserId = tostring(TargetResources[0].id)
+  ) on $left.UserId == $right.NewUserId
   | extend timestamp = StartTime, AccountCustomEntity = UserPrincipalName
 
 entityMappings:
   - entityType: Account
     fieldMappings:
       - identifier: FullName
-        columnName: AccountCustomEntity
\ No newline at end of file
+        columnName: AccountCustomEntity
```

Comment:
Wondering if we should also have an IP Entity in here that might be surfaced through BehaviorAnalytics table.

Reply:
Added, not used yet with hunting queries in the UX but definitely good to have it in place